src/entries/popup/pages/home/Tokens.tsx

```diff
@@ -144,17 +159,18 @@ export function Tokens() {
     },
   );
 
-  const combinedAssets = useMemo(
-    () => [...assets, ...customNetworkAssets],
-    [assets, customNetworkAssets],
-  );
-
   const isPinned = useCallback(
     (assetUniqueId: string) =>
       pinnedAssets.some(({ uniqueId }) => uniqueId === assetUniqueId),
     [pinnedAssets],
   );
 
+  const combinedFilteredAssets = useMemo(
+    () =>
+      [...assets, ...customNetworkAssets].filter((asset) => !isHidden(asset)),
+    [assets, customNetworkAssets, isHidden],
+  );
+
```

Comment:
The naming in this file is getting kind of wonky. We are creating `combinedFilteredAssets` then using that value to create `filteredAssets`.

Maybe we could describe the filter happening in the first value so it's easier to think about later. Something like `unhiddenAssets` or `assetsToDisplay` or `assetsToShow`?

Reply:
Good point. Fixing rn